plugins/event_notification/providers/kafka/lib/model/KafkaNotificationTemplate.php

```diff
@@ -0,0 +1,323 @@
+	const CUSTOM_DATA_MESSAGE_FORMAT = 'messageFormat';
+	const CUSTOM_DATA_API_OBJECT_TYPE = 'apiObjectType';
+	const SCHEMA_ID = 'schemaId';
+	const SCHEMA = 'schema';
+	
+	public function __construct()
+	{
+		$this->setType(KafkaNotificationPlugin::getKafkaNotificationTemplateTypeCoreValue(KafkaNotificationTemplateType::KAFKA));
+		parent::__construct();
+	}
+	
+	public function fulfilled(kEventScope $scope)
+	{
+		if(!kCurrentContext::$serializeCallback)
+			return false;
+		
+		if(!parent::fulfilled($scope))
+			return false;
+		
+		return true;
+	}
+	
+	public function setTopicName($value)
+	{
+		return $this->putInCustomData(self::CUSTOM_DATA_TOPIC_NAME, $value);
+	}
+	
+	public function getTopicName()
+	{
+		return $this->getFromCustomData(self::CUSTOM_DATA_TOPIC_NAME);
+	}
+	
+	public function setPartitionKey($value)
+	{
+		return $this->putInCustomData(self::CUSTOM_DATA_PARTITION_KEY, $value);
+	}
+	
+	public function getPartitionKey()
+	{
+		return $this->getFromCustomData(self::CUSTOM_DATA_PARTITION_KEY);
+	}
+	
+	public function setMessageFormat($value)
+	{
+		return $this->putInCustomData(self::CUSTOM_DATA_MESSAGE_FORMAT, $value);
+	}
+	
+	public function getMessageFormat()
+	{
+		return $this->getFromCustomData(self::CUSTOM_DATA_MESSAGE_FORMAT);
+	}
+	
+	public function setApiObjectType($value)
+	{
+		return $this->putInCustomData(self::CUSTOM_DATA_API_OBJECT_TYPE, $value);
+	}
+	
+	public function getApiObjectType()
+	{
+		return $this->getFromCustomData(self::CUSTOM_DATA_API_OBJECT_TYPE);
+	}
+	
+	public function dispatch(kScope $scope)
+	{
+		KalturaLog::debug("Dispatching event notification with name [{$this->getName()}] systemName [{$this->getSystemName()}]");
+		if(!$scope || !($scope instanceof kEventScope))
+		{
+			KalturaLog::err('Failed to dispatch due to incorrect scope [' . $scope . ']');
+			return;
+		}
+		
+		if(!kConf::hasMap('kafka'))
+		{
+			KalturaLog::debug("Kafka configuration file (kafka.ini) wasn't found!");
+			return;
+		}
+		
+		$object = $scope->getEvent()->getObject();
+		if(!$object)
+		{
+			KalturaLog::debug("Object not found breaking event handling flow");
+			return;
+		}
+		
+		$partitionKey = $this->getPartitionKey();
+		$getter = "get" . ucfirst($partitionKey);
+		if(!is_callable(array($object, $getter)))
+		{
+			KalturaLog::debug("Partition key getter not found on object");
+			return;
+		}
+		
+		$partitionKeyValue = $object->$getter();
+		if(!$partitionKeyValue)
+		{
+			KalturaLog::debug("Partition key value [$partitionKeyValue] empty or not found on object");
+			return;
+		}
+		
+		$modifiedColumns = array();
+		if($scope->getEvent() instanceof kObjectChangedEvent)
+		{
+			$modifiedColumns = $scope->getEvent()->getModifiedColumns();
+			$modifiedColumns = $this->buildMessageOldValues($modifiedColumns);
+		}
+		
+		$apiObjectType = $this->getApiObjectType();
+		$apiObject = call_user_func(kCurrentContext::$serializeCallback, $object, $apiObjectType, 1);
+		
+		$msg = array(
```

Comment:
partnerId will also be a property we will need on those events

Reply:
Partner Id will be on the object itself as our object contain it